go/vt/vtgate/plugin_mysql_server.go

```diff
@@ -550,6 +550,11 @@ func rollbackAtShutdown() {
 		}
 	}()
 
+	if vtgateHandle == nil {
```

Comment:
nit: I'd have done this before starting the goroutine which again checks for nil.

Reply:
You and your logic... 😀

Reply:
https://github.com/vitessio/vitess/pull/10887